Microsoft.Toolkit.Uwp.UI.Controls.Graph/PeoplePicker/PeoplePicker.Events.cs

```diff
@@ -57,88 +71,158 @@ private void ClearAndHideSearchResultListBox()
                     else
                     {
                         ClearAndHideSearchResultListBox();
                     }
                 }
             }
-            catch (Exception)
+            catch (Exception exception)
             {
+                MessageDialog messageDialog = new MessageDialog(exception.Message);
+                await messageDialog.ShowAsync();
             }
             finally
             {
                 IsLoading = false;
             }
         }
 
-        private void SearchResultListBox_OnSelectionChanged(object sender, Windows.UI.Xaml.Controls.SelectionChangedEventArgs e)
+        private void SearchResultListBox_OnSelectionChanged(object sender, SelectionChangedEventArgs e)
         {
-#pragma warning disable SA1119 // Statement must not use unnecessary parenthesis
-            if (!((sender as ListBox)?.SelectedItem is Person person))
-#pragma warning restore SA1119 // Statement must not use unnecessary parenthesis
+            if (sender is ListBox listBox && listBox.SelectedItem is Person person)
             {
-                return;
-            }
+                if (!AllowMultiple && Selections.Any())
+                {
+                    Selections.Clear();
+                    Selections.Add(person);
+                }
+                else
+                {
+                    Selections.Add(person);
+                }
 
-            if (!AllowMultiple && Selections.Any())
-            {
-                Selections.Clear();
-                Selections.Add(person);
+                RaiseSelectionChanged();
+                _searchBox.Text = string.Empty;
             }
-            else
+        }
+
+        private void SelectionsListBox_Tapped(object sender, TappedRoutedEventArgs e)
+        {
+            if (e.OriginalSource is FrameworkElement sourceElement
+                && sourceElement.FindAscendantByName("PersonRemoveButton") is FrameworkElement removeButton
+                && removeButton.Tag is Person person)
             {
-                Selections.Add(person);
+                Selections.Remove(person);
+                RaiseSelectionChanged();
             }
+        }
 
-            RaiseSelectionChanged();
-
-            _searchBox.Text = string.Empty;
+        private void RaiseSelectionChanged()
+        {
+            SelectionChanged?.Invoke(this, new PeopleSelectionChangedEventArgs(Selections));
         }
 
-        private void SelectionsListBox_Tapped(object sender, Windows.UI.Xaml.Input.TappedRoutedEventArgs e)
+        private void SearchBox_OnSizeChanged(object sender, SizeChangedEventArgs e)
         {
-            var elem = e.OriginalSource as FrameworkElement;
+            _searchResultListBox.Width = _searchBox.ActualWidth;
+        }
 
-            var removeButton = elem.FindAscendantByName("PersonRemoveButton");
-            if (removeButton != null)
+        private void SearchResultListBox_OnLayoutUpdated(object sender, object e)
```

Comment:
I think you want to be using a `Flyout` instead no?  As this is just for temporary input selection and then the user can light-dismiss it.  Then you shouldn't need any of these calculations to position a popup as the flyout can be associated with your searchbox.

Reply:
@michael-hawker We figured out a different way to overcome that issue, could you review the updated code with `Flyout`? Thanks.